Implement numeric representation of datetime

ZhenyeLee opened the pull request:
### Description

Before this PR, there is no conversion from numeric to datetime, and DATEADD function doesn't support numeric representation of datetime. The tests are failed in all CONVERT(DATETIME, source_type_value) and DATEADD(source_type_value, source_type_value), also failed in Datetime_proc1  and SMALLDatetime_proc1. When converting BIT to DATETIME implicitly, the test passes when BIT is an Integer number, but fails when BIT is represents as NUMERIC, the reason is we doesn't support implicit convert from NUMERIC to BIT. In this PR, we make casting from NUMERIC to BIT as implicit. This PR also adds numeric representation into ``babelfish_conv_string_to_datetime`` function to convert NUMERIC to DATETIME. The numeric representation pattern starts with an optional minus sign, followed by one or more digits, optionally followed by a decimal point and one or more digits. When the input matches numeric representation pattern, we added this numeric input with 1/1/1900. For DATEADD function, this PR adds a helper function to help adding numeric representation with DATETIME.

Failed tests before this PR:
SELECT CONVERT(DATETIME, source_type_value)
SELECT DATEADD(source_type_value, source_type_value)
Datetime_proc1
SMALLDatetime_proc1

### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -**


* **Boundary conditions -**


* **Arbitrary inputs -**


* **Negative test cases -**


* **Minor version upgrade tests -**


* **Major version upgrade tests -**


* **Performance tests -**


* **Tooling impact -**


* **Client tests -**



### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
Implement numeric representation of datetime

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Update upgrade file

Signed-off-by: Zhenye Li <zhenye@amazon.com>

rishabhtanwar29 committed:
Expose extension_config_remove function through a wrapper (#1314)

Expose extension_config_remove function through a wrapper
so that it can be used by the extensions.

Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>
Co-authored-by: Rishabh Tanwar <ritanwar@amazon.com>

Commit:
Retrigger test

Commit:
Retrigger test

Commit:
Remove from upgrade file

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Add tests

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Add test to mvu file

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Move test to latest

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
remove tests from expected_dependency

Signed-off-by: Zhenye Li <zhenye@amazon.com>

shalinilohia50 committed:
Fix the column "is_ms_shipped" of the view "sys.all_views" (#1312) (#1318)

The column "is_ms_shipped" of the view "sys.all_views" should be correctly set to 0 or 1 based on the view being a system view or a user created view.

Currently, is_ms_shipped is hardcoded to "0" which means SSMS treats all the system views as User defined views which is incorrect.

Task: BABEL-4040

Signed-off-by: Shalini Lohia lshalini@amazon.com

aadityavermaa committed:
Do not dump ROWVERSION/TIMESTAMP column values (#1308)

ROWVERSION/TIMESTAMP column values should not get dumped by pg_dump, instead these columns should be recomputed in the target cluster.

engine repository PR: babelfish-for-postgresql/postgresql_modified_for_babelfish#110
Issues Resolved

BABEL-3819

---------

Signed-off-by: Aditya Verma <adiityav@amazon.com>
Co-authored-by: Aditya Verma <adiityav@amazon.com>

Commit:
Create common upgrade file

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Upgrade common version

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Use Alter in upgrade file

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Drop CAST before change it to implicit

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Add more functions to upgrade file

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Move function from common to tsql

Signed-off-by: Zhenye Li <zhenye@amazon.com>

lejaokri committed:
dev-tools.sh supporting MVU upgrade tests (#1306)

Modified dev-tools.sh to support running MVU test cases manually.

- The dev-tools.sh script has been updated to support running MVU (upgrade) testing.
- A new parameter has been introduced called TEST_MODE : [normal, prepare, verify]. Executing ./dev-tools.sh without any parameter shows more information.
- In order to run the non-MVU tests under input/ directory, the command now becomes ./dev-tools.sh test normal instead of the old way ./dev-tools.sh test input.

Signed-off-by: Kristian Lejao <klejao@amazon.com>

ZhenyeLee committed:
Make DATETIME accepted DATETIME2(7) parameter (#1277)

* Make DATETIME accepted DATETIME2(7) parameter

Before this PR, DATETIME not accepted for DATETIME2(7) parameter and BABELFISH throws error datetime2(7) precision must be between 0 and 6. In our code, we use MAX_TIMESTAMP_PRECISION as 6 as the same as engine side, but DATETIME2 should accept 7 when doing scale. In this PR, we reuse constant value MAX_DATETIME2_PRECISION as 7 in order to let DATETIME accepted DATETIME2(7) parameter.

---------

Signed-off-by: Zhenye Li <zhenye@amazon.com>
Co-authored-by: Zhenye Li <zhenye@amazon.com>

Jakeowen1 committed:
JSON_MODIFY() does not properly handle JSON_QUERY, SELECT FOR JSON, or JSON_MODIFY inputs (#1295)

Currently Babelfish does not properly handle the JSON_MODIFY function when the new value parameter is JSON_QUERY, SELECT FOR JSON, or JSON MODIFY. This change adds support to correctly handle these cases.

Task: Babel-3696
Author: Jacob Owen <owjco@amazon.com>

RIC06X committed:
Fix error message when creating views on temporary tables

Previously when creating views on temporary tables, Babelfish would
throw a general error message which does not explain the reason. This
commit corrects the error message and keeps it the same as in SQL
Server.

Task: BABEL-1645
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

deepakshi-mittal committed:
Fix Create Function cannot contain Commit, Rollback and Execute

Currently, Babelfish allows creating function containing a Commit,
Rollback, exec and execute. Although eoor occurs on executing such
functions. This tasks prevenst such function creation and raises
error message for same.

Task: BABEL-1591

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

RIC06X committed:
Made dbname arg in sp_heapdb case-insensitive

Previously sp_helpdb in babelfish treats dbname as case-sensitive
whereas tsql treats the dbname argument as case-insensitive. This
commit implemented a str_to_lower function for dbname and resol-
ved the case-senstive issue.

Task: BABEL-3643
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

Jakeowen1 committed:
JSON_MODIFY() does not properly handle INT values (#1328)

In TSQL, JSON_MODIFY() is explicitly stated to only accept text or nvarchar inputs 
but in practice it allows certain inputs to be interpreted as non-text types. This pull 
request adds support to properly handle INT values.

Task: BABEL-3697
Signed-off-by: Jake Owen <owjco@amazon.com>

rishabhtanwar29 committed:
Support database-level dump/restore (#1304)

Engine

Added support for dumping a single Babelfish logical database. Changes are
summarized in following bullet points:

    Added a new option --bbf-database-name in both pg_dumpall and pg_dump to
    let user provide the name of logical database to be dumped.
    Fetch all the physical schema names corresponding to the given logical database
    and add all of them into schema_include_patterns list. With this, existing dump
    logic will make sure to only dump all these schemas as well as all the objects
    contained by these schemas.
    Function bbf_selectDumpableTableData takes care of explicitly marking all
    Babelfish catalog table data to be dumped.
    We need to selectively dump the Babelfish catalog table data corresponding to
    given logical database which is being handled by function getCursorForBbfCatalogTableData
    for COLUMN-INSERT and fixCopyCommand for COPY. These function prepares
    query using joins and conditions to filter the data.
    We skip dumping the dbid column as it might conflict with the existing database's dbid
    so we generate the new dbid during restore. It has been handled for both COLUMN-INSERT
    and COPY options:
    a. COPY: Implemented a hook fill_missing_values_in_copyfrom_hook which will use the
    sequence to generate the new dbid.
    b. COLUMN-INSERT: Used existing pre_parse_analyze_hook hook to handle this case.
    Added a new file dumpall_babel_utils.c to implement Babelfish specific logic to
    selectively dump users. New file is needed since existing dump_babel_utils.c can't
    be used with pg_dumpall utility.
    Logins do not belong to a single database so we will not be dumping the logins so it and
    we will only dump default users of the database (dbo, db_owner and db_guest).
    Due to this CREATE and INSERT might fail for certain logins but it should be fine.
    Current implementation does not dump foreign servers and foreign data wrappers as
    these are global objects and does not belong to a single logical database.
    Note: A logical database with certain migration mode (single-db/multi-db) can only be
    restored on the Babelfish server with the same migration mode. Which means a single-db
    database can be restored only on single-db Babelfish server, same thing applies to multi-db.
    Although the builtin databases like master can be restored on any kind of server.

Extension

    Modified existing instance dump-restore github action to work with database-level
    dump-restore as well. We can use configuration file to specify the logical database name.
    Currently we are testing with master database for which a directory has been in upgrade
    folder.
    Implemented fill_missing_values_in_copyfrom_hook to handle the case of generating
    new dbid for COPY. Handled the same for COLUMN-INSERT in pre_parse_analyze_hook.


Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>
Co-authored-by: Rishabh Tanwar <ritanwar@amazon.com>

deepakshi-mittal committed:
Fix: Corrected compatibility level returned by sp_helpdb

Added test cases for same
Task: BABEL-4029

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

pratikzode committed:
Support STDEV(), STDEVP(), VAR(), VARP() statistical T-SQL aggregates

STDEV()
STDEV function is used in SQL to calculate the standard deviation of a set of values in a given column of a table. STDEV stands for "standard deviation," which is a statistical measure that represents the degree of variation of a set of values from its mean or average

STDEVP()
STDEVP function is similar to the STDEV function in SQL, but it calculates the standard deviation of an entire population rather than a sample of the population. "STDEVP" stands for "standard deviation of a population".

VAR()
VAR function is used in SQL to calculate the variance of a set of values in a given column of a table. "VAR" stands for "variance," which is a statistical measure that represents how much a set of values varies from its mean or average.

VARP()
VARP function is used in SQL to calculate the variance of an entire population rather than a sample of the population. "VARP" stands for "variance of a population"

Task: BABEL-745
Signed-off-by: pratikzode <pratik.zode@gmail.com>

deepakshi-mittal committed:
Create function cannot contain a SAVE, PRINT, WAITFOR, RAISERROR

SAVE TRANSACTION
WAITFOR DELAY
PRINT
RAISERROR

Task: BABEL-1591

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

aadityavermaa committed:
Fix test failures in sp_tablecollations_100 and sys.sql_modules after restore  (#1332)

* BABEL-4016, BABEL-4017

The datatype of column 'name' in the output of sp_tablecollations_100 procedure changed from sys.sysname to text after performing dump/restore on latest->latest path. Further, cache looked failed error was observed when sys.sql_modules was called after dump/restore on the same upgrade path. Both these issues led to failures in tests, which this PR aims to resolve.

Signed-off-by: Aditya Verma <adiityav@amazon.com>

---------

Signed-off-by: Aditya Verma <adiityav@amazon.com>
Co-authored-by: Aditya Verma <adiityav@amazon.com>

Deepesh125 committed:
Fix sp_database test cases (#1342)

Previously, we were having test cases for sp_databases which also test the database size. But database size is subject to change, it will cause the test failure. So this commit updates the test cases to not check the database size.

Task: None
Signed-off-by: Dipesh Dhameliya <dddhamel@amazon.com>

Commit:
Implement CONTEXT_INFO() and SET CONTEXT_INFO (#1288)

Description
===========
CONTEXT_INFO is a session local parameter of type varbinary(128). It can be set
by 'SET CONTEXT_INFO 0x123456' and can be get by 'SELECT CONTEXT_INFO()'. It can
also be get with sys.sysprocesses, sys.dm_exec_requests, sys.dm_exec_sessions
views.

Solution
========
CONTEXT_INFO is similar to HOST_NAME/LANGUAGE, all of them are session local.
The difference is that CONTEXT_INFO supports int/numeric/float parameter and
dose implicit type casts.
We add a procedure bbf_set_context_info(IN context_info sys.VARBINARY(128)) and
rewrite 'SET CONTEXT_INFO 0x123456' to
'CALL bbf_set_context_info(convert(varbinary(128), 0x123456))'.

Limitation
==========
Some type casts are not supported in Babelfish, such as type cast from numeric
to varbinary. Some type casts have their own rules, e.g. integer constant
greater than 2,147,483,647 is converted to decimal instead of bigint.
They are not the issue of CONTEXT_INFO, so are not solved in this patch.

forestkeeper committed:
Add support for rowcount in execution (#1331)

* Add support for rowcount in execution

previously we don't allow set rowcount other than 0
in this commit, we'll support rowcount > 0 for
SELECT/UPDATE/INSERT/DELETE

Signed-off-by: Zhibai Song <szh@amazon.com>
Co-authored-by: Zhibai Song <szh@amazon.com>

pratikzode committed:
BABEL-3478 Support the ROWCOUNT_BIG() function. (#1343)

* BABEL-3478 Support the ROWCOUNT_BIG() function.

We have implemented function rowcount_big()

Task: BABEL-3802
Signed-off-by: pratikzode <pratik.zode@gmail.com>

KushaalShroff committed:
Long TVP PARAMETER names in RPC throw error (#1346)

ISSUE: We create a unique temp-table-name by adding the following suffix: "{}TDSTVP_TEMP_TABLE{_}%d", (int)rand() Now if we create a temp table with this unique temp-table-name (however long), it gets created. But we have a tvp_lookup_list which maintains the list of tvps the TSQL Extension has to read. For that list we have inserted a modified and truncated TVP name. Rather than storing tableName we store finalTableName, which CAN be different from what was created. Although the though behind it seems logical but it(truncation) should have been done for the temp-table which got created.

FIX: So to fix this we had to truncate the actual tvp name along with its schema name, if provided. We also had to move the logic to fetch tvp type oid from ReadParameter - fetch phase to Recv Function - process phase because we required the truncated names to do so. NOTE: We cannot do truncation in Fetch phase because we require 1. A transaction 2. MessageMemoryContext which are both available during Process phase

Signed-off-by: Kushaal Shroff kushaal@amazon.com

bsatarupa committed:
DATEADD() gives wrong output with DATETIMEOFFSET (#1340)

Postgres outputs datetimeoffset data by removing/subtracting input timezone value from it.
Fix includes, working on the engine output to include timezone and thereby retain the input timestamp as is.

Task: BABEL-1625

Signed-off-by: Satarupa Biswas <satarupb@amazon.com>
Co-authored-by: Satarupa Biswas <satarupb@amazon.com>

pratikzode committed:
BABEL-1199 Support DATABASE_PRINCIPAL_ID() T-SQL function (#1344)

* BABEL-1199 Support DATABASE_PRINCIPAL_ID() T-SQL function

SELECT DATABASE_PRINCIPAL_ID();
Retrieve the ID of the current user.

SELECT DATABASE_PRINCIPAL_ID(‘principal_name’);
Retrieve the ID of the specific database principal.

Task: BABEL-1199
Signed-off-by: pratikzode <pratik.zode@gmail.com>

* BABEL-1199 Added to schedule file

Have added BABEL-1199 to 13_6, 13_9, 14_3, 14_5, 14_6, 14_8,15_1,15_2

Task: BABEL-1199
Signed-off-by: pratikzode <pratik.zode@gmail.com>

* BABEL-1199 Deleted test from schedule

Task: BABEL-1199
Signed-off-by: pratikzode <pratik.zode@gmail.com>

* BABEL-1199 Added sys schema

Task: BABEL-1199
Signed-off-by: pratikzode <pratik.zode@gmail.com>

---------

Signed-off-by: pratikzode <pratik.zode@gmail.com>

RIC06X committed:
Fix uninitialized error on context_info

Task: BABEL-748

Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

lejaokri committed:
Include public, sys, and INFORMATION_SCHEMA in sys.database_principals VIEW

These dummy principals for read-only purposes. There are no actual objects behind them.

Task: BABEL-3230

Signed-off-by: Kristian Lejao <klejao@amazon.com>

thephantomthief committed:
Fix crash in sp_addlinkedserver (#1345)

This commit fixes a crash in sp_addlinkedserver that was arising due
to an attempt to free pointer to a string literal.

Signed-off-by: Sharu Goel goelshar@amazon.com

2jungkook commented on contrib/babelfishpg_tsql/sql/sys_functions.sql:
```diff
@@ -1141,6 +1178,46 @@ END;
+        digit_to_startdate := CAST('1900-01-01 00:00:00.0' AS sys.DATETIME) + CAST(startdate as sys.DATETIME);
+    END IF;
+
+    CASE datepart
```
Please add test cases for all possible branches here.

2jungkook commented on test/JDBC/upgrade/latest/schedule:
```diff
@@ -268,6 +268,7 @@ BABEL-SP_SPROC_COLUMNS_100-dep
 BABEL-3000-dep
 Test-sp_helprole-dep
 Test-sp_helprolemember-dep
+TestDatetime-sql
```
In addition to the latest schedule file, please add appropriate tests to at least 15.2 schedule file. You may need to split some tests (one for the latest and another for the previous versions).

Deepesh125 committed:
Run pgindent on each babelfish extensions code (#1357)

* Run pgindent on each babelfish extensions

* fix compilation error

---------

Co-authored-by: Dipesh Dhameliya <dddhamel@amazon.com>

aadityavermaa committed:
add spt_tablecollations_view to upgrade script (#1341)

The changes made in #1332 also require adding the sys.spt_tablecollations_view view into the latest upgrade script since the view definition is being changed. In this PR, the view is added to 3.1.0-3.2.0 upgrade script.

Signed-off-by: Aditya Verma <adiityav@amazon.com>
Co-authored-by: Aditya Verma <adiityav@amazon.com>

Deepesh125 committed:
Add 243e3294c5fee7863e7846b30d8a821b7e4b4242 to .git-blame-ignore-revs (#1360)

Authored-by: Dipesh Dhameliya <dddhamel@amazon.com>

skkyal committed:
Create scalar UDFs as STABLE functions by default (#1269)

Earlier the tsql functions were created with volatility as volatile by default. From behaviour and performance perspective function should be created as 'stable' by default. Also, When a function is stable, it cannot have DDL or DML inside it. But it possible to DML only in table variables in tsql.
With this PR made the default volatility during tsql function creation as stable except for the cases when a function contains a table variable (tvp or tvf).

Signed-off-by: Shlok Kumar Kyal (skkyal@amazon.com)

bsatarupa committed:
Fixing existing expected file due to changes added as part of BABEL-1625 (#1364)

Postgres outputs datetimeoffset data by removing/subtracting input timezone value from it.
Datetimeoffset change in PR #1340 works on the engine output to include timezone and thereby retain the input timestamp as is.

In this PR, we have fixed errors in existing expected file generated due to changes added to BABEL-1625.

Task: BABEL-1625
Signed-off-by: Satarupa Biswas satarupb@amazon.com

hash-16 committed:
Changes to support DDL Export Test Framework (#1326)

Adding support for DDL test framework into python framework.
We in the framework are comparing the generated exported DDL export from the BBF and to the expected exported DDL from the sql server instance.

Issues Resolved
[BABEL-3969]

Signed-off-by: Ashish Prasad pashisht@amazon.com

thephantomthief committed:
Add build instructions for linked servers and disable them by default (#1369)

This commit disables the linked servers feature by default and only keeps it enabled in the github actions. The instructions to enable the feature is added to the build instructions of the Babelfish extensions.

Task: BABEL-OSS
Signed-off-by: Sharu Goel <goelshar@amazon.com>

hash-16 committed:
Disable DDl testcase (#1374)

Disable DDL test cases from python framework , taking lot of time to execute

Signed-off-by: Ashish Prasad pashisht@amazon.com

Jakeowen1 committed:
Created macro for PGC_SUSET (#1362)

This commit creates a macro for (superuser() ? PGC_SUSET : PGC_USERSET) named GUC_CONTEXT_CONFIG to avoid future merge conflicts.

Task: BABEL-OSS

Signed-off-by: Jake Owen <owjco@amazon.com>

Commit:
Fix the issue that context_info() cannot display complete varbinary in sqlcmd. (#1355)

Description
===========
context_info() cannot display complete varbinary in sqlcmd.

Analysis
========
Based on the TDS document, it is expected that sqlcmd would have the capability
to support both variable-length and partially length-prefixed data types.
However, in this particular scenario, sqlcmd does not support the partially
length-prefixed data type for varbinary.
The main distinguishing factor between these two data types is that the
variable-length data type has an associated length value, while the
partially length-prefixed data type does not require the full data length to be
specified before the actual data is streamed out.

Note: Neither freetds nor JDBC encounter any issues with regard to this matter.

Fix
===
Fill atttypmod in PrepareRowDescription with the typmod value stored in
pg_proc.probin.

basasairohan committed:
Add manual and scheduled workflow for generating Code Coverage (#1370)

This change adds a code-coverage workflow which is scheduled to run at 00:00:00 UTC every day and it can also be triggered manually by specifying the branch name as an input.

Scheduled workflow generates code coverage for BABEL_3_X_DEV and BABEL_2_X_DEV branches.
We can also trigger the manual workflow for BABEL_3_X_DEV
Task: TES2-12
Signed-off-by: Sai Rohan Basa bsrohan@amazon.com

pratikzode committed:
TES2_14 Moving Test to JDBC (#1367)

* TES2_14 Moving Test to JDBC

Moved
babel_219,
babel_collection,
babel_datatype,
babel_ddl,
babel_delete1,
babel_function,
babel_like,
babel_typecode,
babel_uniqueidentifier
babel_set_command
babel_table_type
babel_transaction
babel_emoji
to JDBC
And removed it from babelfish_extensions/contrib/babelfishpg_tsql/sql/test directory

Signed-off-by: pratikzode <pratik.zode@gmail.com>

aadityavermaa committed:
Integrate code coverage steps in dev-tools.sh  (#1368)

In this change, steps to generate code coverage report are integrated in dev-tools.sh so that developers can use it to get code coverage on their dev-desktops.

Task: TES2-12
Signed-off-by: adiityav@amazon.com

Deepesh125 committed:
Extend functionality of dev-tools.sh to run pgindent also (#1375)

Extend functionality of dev-tools.sh to run pgindent also

Task: No-jira
Signed-off-by: Dipesh Dhameliya <dddhamel@amazon.com>

hash-16 committed:
Optimizing DDL Export Test Framework (#1377)

Added two different ways of scripting the objects which can be switched by passing flag to the script according to the input file name.

Issues Resolved
[BABEL-3969]

Signed-off-by: Ashish Prasad pashisht@amazon.com

pratikzode committed:
Support DATABASE_PRINCIPAL_ID() T-SQL function

SELECT DATABASE_PRINCIPAL_ID();
Retrieve the ID of the current user.

SELECT DATABASE_PRINCIPAL_ID(‘principal_name’);
Retrieve the ID of the specific database principal.

Modified the file so that when called with NULL as argument it will
return NULL and also modified for trailing spaces that will give proper
output insted of NULL.

Task: BABEL-1199
Signed-off-by: pratikzode <pratik.zode@gmail.com>

forestkeeper committed:
In CTE Null first order behavior is not correct (#1382)

* Solve the bug in CTE Null first order behavior is not correct

Previously we add the Null first check after analyze, and in that case
it'll miss to add the Null first tag to the sort clause for a CTE query

change the place adding Null first tag to the place sortClause is
initialized

Signed-off-by: Zhibai Song <szh@amazon.com>
Task: BABEL-3991

pratikzode committed:
TES2_14 Moving Test to JDBC (#1386)

* TES2_14 Moving Test to JDBC

Moved
babel_219,
babel_collection,
babel_datatype,
babel_ddl,
babel_delete1,
babel_function,
babel_like,
babel_typecode,
babel_uniqueidentifier
babel_set_command
babel_table_type
babel_transaction
babel_emoji
to JDBC
And removed it from babelfish_extensions/contrib/babelfishpg_tsql/sql/test directory

Modified babel_transaction file.
Forget to add Go in last few lines

Task: TES2_14
Signed-off-by: pratikzode <pratik.zode@gmail.com>

Commit:
Add tests before 15.3

Signed-off-by: Zhenye Li <zhenye@amazon.com>

Deepesh125 committed:
Fixed test case failure related to sp_databases (#1378)

Output of the system procedure, sp_databases contains the size of the database which is subjected to vary. Hence, it was causing the test failures unexpectedly.

This commit adds a minor improvement to the definition of the view sys.sp_databases_view to consider only physical relations to calculate the database size. We are temporarily disabling the test case for sp_databases to fix the test flakiness arising due to database size.

Task: BABEL-4052
Signed-off-by: Dipesh Dhameliya <dddhamel@amazon.com>

Commit:
Correct test output file

Signed-off-by: Zhenye Li <zhenye@amazon.com>

ZhenyeLee committed:
Merge branch 'BABEL_3_X_DEV' into jira-BABEL-323
